### PR TITLE
[860qfetxf][base-triger] Feature/clear hint to filter trigger extends

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
Added composition api for FilterTrigger.

## How has this been tested?
Manually + unit-test

## Screenshots (if appropriate):
Screenshot 2024-04-09 at 21 31 14

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [X] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/base-trigger/src/FilterTrigger.jsx [hunk 1]
```diff
@@ -101,51 +127,105 @@ class RootFilterTrigger extends Component {
...
-              onClick={callAllEventHandlers(onClear, this.handleClear, this.handleStopPropagation)}
-              onKeyDown={this.handleStopPropagation}
-              disabled={disabled}
-              id={`igc-${uid}-filter-trigger-clear`}
-              aria-labelledby={`igc-${uid}-filter-trigger-clear ${id}`}
-              aria-label={getI18nText('clear')}
-            >
-              <FilterTrigger.Addon tag={Close} />
-            </SFilterTrigger>
+          {TriggerButton ?? (
```
Comment (msereniti) [inline semcore/base-trigger/src/FilterTrigger.jsx:130]: 1. Do I understand correctly that I can't render any children without TriggerButton?
2. Is it possible to wrap TriggerButton? For example, add tooltip over TriggerButton?

Comment (ilyabrower) [reply semcore/base-trigger/src/FilterTrigger.jsx:130]: 1. Yes, but this is current behaviour too. All your children will be rendered inside TriggerButton. But you shouldn't wrap them into TriggerButton, if you don't need to use custom clear button.
2. No. You could wrap all FilterTrigger or Content inside TriggerButton. For example,
``

File: semcore/utils/src/findComponent.ts [hunk 2]
```diff
@@ -6,27 +6,34 @@ import getOriginChildren from './getOriginChildren';
 function findComponent(
   Children: any,
   names: string[],
+  recursively?: boolean,
```
Comment (msereniti) [inline semcore/utils/src/findComponent.ts:9]: Could you add some basic recursion depth protection?

**Review by msereniti — APPROVED** [review]

File: semcore/utils/src/findComponent.ts (hunk 2 above)
Comment (ilyabrower) [reply semcore/utils/src/findComponent.ts:9]: added

**Pull request merged**